torch/utils/waitcounterfilelock.py

```diff
@@ -0,0 +1,37 @@
+from types import TracebackType
+from typing import Self
+
+from filelock import FileLock as base_FileLock
+
+from torch.monitor import _WaitCounter
+
+
+class WaitCounterFileLock(base_FileLock):
```

Comment:
I hate the name on this file. Just call this FileLock, the wait counter is an incidental detail, right?

In fact, we could potentially just inline filelock implementation here and dump that package entirely (yay less deps)

Reply:
name updated.

Re: less deps, my understanding is pytorch works on multiple os's correct? Most of the complexity in the underlying filelock is supporting multiple operating systems.